src/Microsoft.Metadata.Visualizer/ILVisualizer.cs

```diff
@@ -417,11 +431,33 @@ public void VisualizeHeader(StringBuilder sb, int codeSize, int maxStack, Immuta
                             break;
 
                         case OperandType.ShortInlineR:
-                            sb.AppendFormat(" 0x{0:x8}", ReadUInt32(ilBytes, ref curIndex));
+                            {
+                                var value = ReadSingle(ilBytes, ref curIndex);
+                                if (value == 0 && 1 / value < 0)
+                                {
+                                    sb.Append(" -0.0");
+                                }
+                                else
+                                {
+                                    sb.Append(" ");
+                                    sb.Append(VisualizeSingle(value));
```

Comment:
This is odd - to be able to override the behavior for some values but not others. Why not move this logic to `VisualizeSingle()`?